client/components/forms/form-radio-with-thumbnail/index.jsx

```diff
@@ -0,0 +1,44 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import FormRadio from 'components/forms/form-radio';
+import FormLabel from 'components/forms/form-label';
+import cssSafeUrl from 'lib/css-safe-url';
+
+const FormRadioWithThumbnail = ( { label, thumbnail, ...otherProps } ) => {
+	const { cssClass, cssColor, imageUrl } = thumbnail;
+	const styles = {
+		backgroundColor: cssColor,
+		backgroundImage: imageUrl ? 'url(' + cssSafeUrl( imageUrl ) + ')' : '',
+	};
```

Comment:
Thanks for chiming in, @folletto!
It depends whether we interpret the image as content or decoration. Depending on the use-case one could argue for one or the other.

Semantics aside though, I think there's a valid reason for using a background image here from a visual/technical point of view:

We want the component to be as flexible/reusable as possible. If we consider it being used in the context of user-generated images, then we have less control over which images are being provided to the component.

This is where a `background-image` with `background-size: cover` provides us with a high likelihood of a visually appealing result regardless of the image dimensions:

<img width="421" alt="background-size-cover" src="https://user-images.githubusercontent.com/1562646/29536256-6330af08-86bd-11e7-9669-a48ee2b43c4c.png">

If we were to use an `<img />` element, we could, of course, achieve basically the same result in modern browsers with `object-fit: cover`:

<img width="417" alt="object-size-cover" src="https://user-images.githubusercontent.com/1562646/29536328-a5d7cc10-86bd-11e7-8985-2a18e99aafaf.png">

But as we still have to support IE11 and Edge, we have the issue that these browsers don't support the `object-fit` property, yet. This results in squashed images:

<img width="418" alt="object-size-cover-ie" src="https://user-images.githubusercontent.com/1562646/29536443-26c307a4-86be-11e7-98ca-b68250bde938.png">

This means we'd have to add a feature detection test for `object-fit` and then polyfill the behaviour of `object-fit` by using a background image on the parent element that is scaled using `background-size: cover`. 

I'd argue that a minor semantic improvement doesn't really justify adding an additional feature detection test and a polyfill just to replicate behaviour we could get out of the box when using a background image by default.

Of course, we could also use neither `background-size: cover` nor `object-fit: cover` and just scale the image to 100% width. This impacts the visual appearance of course:

<img width="416" alt="just-img-100 -width" src="https://user-images.githubusercontent.com/1562646/29536785-7d01839c-86bf-11e7-9d3d-a56d4f43db62.png">

Based on the description above I'd advocate for continuing to use a background image with `background-size: cover` here. But I'm of course open to change it if the semantics are of higher priority. In that case please advise which solution to go for.